packages/web-components/fast-foundation/src/di/di.ts

```diff
@@ -0,0 +1,1414 @@
+        }
+    }
+    return clone;
+}
+
+/**
+ * @alpha
+ */
+export class ResolverBuilder<K> {
+    public constructor(private container: Container, private key: Key) {}
+
+    public instance(value: K): Resolver<K> {
+        return this.registerResolver(ResolverStrategy.instance, value);
+    }
+
+    public singleton(value: Constructable): Resolver<K> {
+        return this.registerResolver(ResolverStrategy.singleton, value);
+    }
+
+    public transient(value: Constructable): Resolver<K> {
+        return this.registerResolver(ResolverStrategy.transient, value);
+    }
+
+    public callback(value: ResolveCallback<K>): Resolver<K> {
+        return this.registerResolver(ResolverStrategy.callback, value);
+    }
+
+    public cachedCallback(value: ResolveCallback<K>): Resolver<K> {
+        return this.registerResolver(
+            ResolverStrategy.callback,
+            cacheCallbackResult(value)
+        );
+    }
+
+    public aliasTo(destinationKey: Key): Resolver<K> {
+        return this.registerResolver(ResolverStrategy.alias, destinationKey);
+    }
+
+    private registerResolver(strategy: ResolverStrategy, state: unknown): Resolver<K> {
+        const { container, key } = this;
+        this.container = this.key = (void 0)!;
+        return container.registerResolver(key, new ResolverImpl(key, strategy, state));
+    }
+}
+
+const dependencyLookup = new Map<Constructable | Injectable, Key[]>();
+
+/**
+ * @alpha
+ */
+export type ParentLocator = (owner: any) => Container | null;
+
+const DILocateParentEventType = "__DI_LOCATE_PARENT__";
+const defaultFriendlyName = "(anonymous)";
+
+/**
+ * @alpha
+ */
+export interface DOMParentLocatorEventDetail {
+    container: Container | void;
+}
+
+function domParentLocator(element: HTMLElement): Container {
+    const event = new CustomEvent<DOMParentLocatorEventDetail>(DILocateParentEventType, {
+        bubbles: true,
+        composed: true,
+        cancelable: true,
+        detail: { container: void 0 },
+    });
+
+    element.dispatchEvent(event);
+
+    return event.detail.container || DI.getOrCreateDOMContainer();
+}
+/**
+ * @alpha
+ */
+export interface InterfaceConfiguration {
+    /**
+     * The friendly name for the interface. Useful for debugging.
+     */
+    friendlyName?: string;
+
+    /**
+     * When true, the dependency will be re-resolved when FASTElement connection changes.
+     * If the resolved value changes due to connection change, a {@link @microsoft/fast-element#Observable.notify | notification }
+     * will be emitted for the property, with the previous and next values provided to any subscriber.
+     */
+    respectConnection?: boolean;
+}
+
+function createInterface<K extends Key, T = any>(
+    friendlyName?: string
+): DefaultableInterfaceSymbol<K, T>;
+function createInterface<K extends Key, T = any>(
+    configuration?: InterfaceConfiguration & { respectConnection: false }
+): DefaultableInterfaceSymbol<K, T>;
```

Comment:
@EisenbergEffect All of my changes to make re-resolution of DI interface after connection change are in this function.